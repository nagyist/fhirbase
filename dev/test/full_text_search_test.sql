SET escape_string_warning=off;

\set cfg '{"base":"https://test.me"}'
\set alert `cat test/fixtures/alert.json`

BEGIN;

  SELECT assert_eq (
    '(bone | liver) & metastases',
    _text_to_query('(boNe OR liveR) and metastases'),
    'convert text to query');

  WITH alert AS (
    SELECT *
    FROM fhir_create(:'cfg', 'Alert', :'alert'::jsonb, '[]'::jsonb) as bundle
  ), searching AS (
    SELECT a.*,
      fhir_search(:'cfg', 'Alert', '_text=%28previous%20and%20active%20and%20note%29%20OR%20absent_text') as present,
      fhir_search(:'cfg', 'Alert', '_text=absent_text') as absent
    FROM alert a
  )
  SELECT
    assert_eq(
      s.bundle#>>'{entry,0,id}',
      s.present#>>'{entry,0,id}',
      'full text search found'),
    assert_eq(
      0,
      jsonb_array_length(s.absent#>'{entry}'),
      'full text search not found')
  FROM searching s;

ROLLBACK;
