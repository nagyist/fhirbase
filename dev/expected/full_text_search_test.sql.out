SET escape_string_warning=off;
SET
BEGIN;
BEGIN
SELECT assert_eq (
    '(bone | liver) & metastases',
    _text_to_query('(boNe OR liveR) and metastases'),
    'convert text to query');
        assert_eq         
--------------------------
 OK convert text to query
(1 row)

WITH alert AS (
    SELECT *
    FROM fhir_create('{"base":"https://test.me"}', 'Alert', '{
  "resourceType": "Alert",
   "status": "active",
   "subject": {
     "reference": "Patient/example",
     "display": "Peter Patient"
   },
   "note": "previous-note"
}'::jsonb, '[]'::jsonb) as bundle
  ), searching AS (
    SELECT a.*,
      fhir_search('{"base":"https://test.me"}', 'Alert', '_text=%28previous%20and%20active%20and%20note%29%20OR%20absent_text') as present,
      fhir_search('{"base":"https://test.me"}', 'Alert', '_text=absent_text') as absent
    FROM alert a
  )
  SELECT
    assert_eq(
      s.bundle#>>'{entry,0,id}',
      s.present#>>'{entry,0,id}',
      'full text search found'),
    assert_eq(
      0,
      jsonb_array_length(s.absent#>'{entry}'),
      'full text search not found')
  FROM searching s;
         assert_eq         |           assert_eq           
---------------------------+-------------------------------
 OK full text search found | OK full text search not found
(1 row)

ROLLBACK;
ROLLBACK
